src/oauth/consumer.cr

```diff
@@ -63,7 +63,8 @@ class OAuth::Consumer
                  @request_token_uri : String = "/oauth/request_token",
                  @authorize_uri : String = "/oauth/authorize",
                  @access_token_uri : String = "/oauth/access_token",
-                 @tls : Bool | OpenSSL::SSL::Context::Client = @scheme == "https")
+                 @tls : Bool | OpenSSL::SSL::Context::Client = false)
+    @tls = @scheme == "https" unless @tls.is_a?(OpenSSL::SSL::Context::Client)
```

Comment:
I'd suggest positive check instead in the vein of `@scheme == "https" if @tls.is_a?(Bool)`.